### ISSUE TITLE
Improving effect plugin selector dialog

### ISSUE THREAD

**Issue by unfa** [LMMS/lmms#3799]
I often would like to be able to manage a list or favourite plugins, - I could find what I need much faster this way.

Maybe there are some other cool things that could be done there like a tab that tracks what plugins do I add most often and give me quick access to those aswell:

![image](https://user-images.githubusercontent.com/491704/30231303-96cce6ac-94ea-11e7-8ee6-aa4980795f1a.png)
![image](https://user-images.githubusercontent.com/491704/30231309-9e4da5ce-94ea-11e7-9fae-e3767342b02b.png)
![image](https://user-images.githubusercontent.com/491704/30231312-a0e73732-94ea-11e7-801a-7f9fadb89815.png)

What do you think?

**Comment by BaraMGB** [comment]
I would like to see all plugins in a browser in our left sidebar.

**Comment by budislav** [comment]
@BaraMGB I made this mockup long time ago as a part of single window.
![ladspa](https://user-images.githubusercontent.com/3619927/30236534-053af24c-951c-11e7-954a-db4df23e2825.png)

**Comment by unfa** [comment]
Would this also handle effect plugin presets?

I see the left-side browser idea as a long-term solution.

But maybe a quick and dirty Favourites tab would solve the problem for a time being before that major rehaul is going to take place?

**Comment by budislav** [comment]
> Would this also handle effect plugin presets?

Yes just under that arrow you can open and see all presets there. This is I think how it works in A. Live. I found this to be more intuitive way. You can insert an instrument or his preset from same place quickly.

**Comment by musikBear** [comment]
> What do you think?

I generally like the idea. but it is on the other hand not something i have missed. The current token-search is very efficient, so in some respect, this would be a luxurious feature

**Comment by tresf** [comment]
I can confirm with @unfa that I spend a bit of time searching for the right plugin and then setting the right presets and often find myself using a similar stack of presets when achieving a particular sound.  Recent and favorite would be nice.  (Never a fan of "favorite", personally, but similar concept).

**Comment by Spekular** [comment]
Personally the biggest improvement for me would be for the search field to
grab focus so I can open Add FX and immediately type, or perhaps scroll to
plugins starting with *X* when typing outside the search field.

**Comment by musikBear** [comment]
Is this best fixed with savable / loadable  FX-chains, as a new file-*type* ?

**Comment by tresf** [comment]
> Is this best fixed with savable / loadable FX-chains, as a new file-type ?

No, it's more of a "commonly used" or "recently used" feature.  Presets are nice and can serve as a temporary solution to this, but that's a separate feature.   Presets are more important to workflow which takes a considerable amount of time to get the settings just right.  (e.g. Stacking or adjusting knobs precisely).   On the contrary, this feature is more like the "Quick Access" or "Recently Opened" lists you'll find in other UIs.
